### PR TITLE
[ANCHOR-148] Generate db migration file and add docker-compose sep24 e2e tests

### PR DESCRIPTION
**Pull request by stfung77**
### What

- generate db migration file
- add docker-compose sep24 e2e tests

### CODE REVIEW

**Review by lijamie98 — APPROVED** [review]

File: platform/src/main/resources/db/migration/V6__sep24_field_updates.sql [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+ALTER TABLE sep24_transaction ADD refunded BOOLEAN;
+
+ALTER TABLE sep24_transaction ADD refunds JSON;
+
+ALTER TABLE sep24_transaction ADD request_asset_code VARCHAR(255);
+
+ALTER TABLE sep24_transaction ADD request_asset_issuer VARCHAR(255);
+
+ALTER TABLE sep24_transaction ADD required_info_message VARCHAR(255);
```
Comment (Ifropc) [inline platform/src/main/resources/db/migration/V6__sep24_field_updates.sql:21]: @lijamie98

File: platform/src/main/resources/db/migration/V6__sep24_field_updates.sql [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+ALTER TABLE sep24_transaction DROP COLUMN started_at;
+
+ALTER TABLE sep24_transaction ADD completed_at TIMESTAMP WITHOUT TIME ZONE;
+
+ALTER TABLE sep24_transaction ALTER COLUMN  id SET NOT NULL;
+
+ALTER TABLE sep31_transaction DROP COLUMN refunds;
+
+ALTER TABLE sep31_transaction ADD refunds JSON;
```
Comment (Ifropc) [inline platform/src/main/resources/db/migration/V6__sep24_field_updates.sql:55]: I think we should copy data from this column to a new (historic) table for bitso
```suggestion
CREATE TABLE sep31_historic (
   refunds VARCHAR(255)
);

INSERT INTO sep31_historic (refunds)
SELECT refunds
FROM sep31_transaction;

ALTER TABLE sep31_transaction DROP COLUMN refunds;

ALTER TABLE sep31_transaction ADD refunds JSON;
```

**Review by Ifropc — CHANGES_REQUESTED** [review]

File: platform/src/main/resources/db/migration/V6__sep24_field_updates.sql (hunk 1 above)
Comment (lijamie98) [reply platform/src/main/resources/db/migration/V6__sep24_field_updates.sql:21]: This field should not be in `JdbcSep24Transaction`. 

We should move the field out from the JdbcSepTransaction to JdbcSep31Transaction and regenerate the migration file.

Comment (lijamie98) [reply platform/src/main/resources/db/migration/V6__sep24_field_updates.sql:21]: .

**Review by lijamie98 — CHANGES_REQUESTED** [review]
Mark request changes to block merge.

**Review by Ifropc — APPROVED** [review]

File: platform/src/main/resources/db/migration/V6__sep24_field_updates.sql (hunk 2 above)
Comment (Ifropc) [reply platform/src/main/resources/db/migration/V6__sep24_field_updates.sql:53]: It's just to keep data that was in refunds table. With the SQL we currently have, we just drop this column entirely.

**Comment by lijamie98** [comment]
I am blocking this PR because we need to add `bank_account_type` column. I will approve after the newly added column is resolved.

File: platform/src/main/resources/db/migration/V6__sep24_field_updates.sql [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+ALTER TABLE sep24_transaction DROP COLUMN completed_at;
+
+ALTER TABLE sep24_transaction DROP COLUMN started_at;
+
+ALTER TABLE sep24_transaction ADD completed_at TIMESTAMP WITHOUT TIME ZONE;
+
+ALTER TABLE sep24_transaction ALTER COLUMN  id SET NOT NULL;
+
+ALTER TABLE sep31_transaction DROP COLUMN refunds;
```
[the remark this answers is not in the document]
Comment (lijamie98) [reply platform/src/main/resources/db/migration/V6__sep24_field_updates.sql:51]: I will do.

File: platform/src/main/resources/db/migration/V6__sep24_field_updates.sql (hunk 2 above)
Comment (lijamie98) [reply platform/src/main/resources/db/migration/V6__sep24_field_updates.sql:53]: Hey @Ifropc This is lovely and awesome!!! 💯 

💡 Is it possible to check if this SQL script work?

Comment (Ifropc) [reply platform/src/main/resources/db/migration/V6__sep24_field_updates.sql:53]: Should work, but would be nice to manually test it with some data in postgres

Comment (lijamie98) [reply platform/src/main/resources/db/migration/V6__sep24_field_updates.sql:65]: Hey @Ifropc I have merged your suggestions to the PR. Can you check if this takes care of the Jira ticket? https://stellarorg.atlassian.net/browse/ANCHOR-171

Comment (Ifropc) [reply platform/src/main/resources/db/migration/V6__sep24_field_updates.sql:65]: Yes, it's an alternative way to resolve this ticket. The original suggestion was to migrate data, with this new table there is no need for migration. Instead, historic records are stored in the separate table

File: platform/src/main/resources/db/migration/V6__sep24_field_updates.sql [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+ALTER TABLE sep24_transaction ADD CONSTRAINT pk_sep24_transaction PRIMARY KEY (id);
+
+CREATE TABLE sep31_historic (
+    refunds VARCHAR(255)
+);
+
+INSERT INTO sep31_historic (refunds)
+SELECT refunds
+FROM sep31_transaction;
```
Comment (Ifropc) [inline platform/src/main/resources/db/migration/V6__sep24_field_updates.sql:61]:
```suggestion
CREATE TABLE sep31_historic (
    id VARCHAR(255) NOT NULL.
    refunds VARCHAR(255)
);

INSERT INTO sep31_historic (id, refunds)
SELECT id, refunds
FROM sep31_transaction;

ALTER TABLE sep31_historic ADD CONSTRAINT FK_SEP31_REFUND FOREIGN KEY (id) REFERENCES sep31_transaction (id);

```

Add id field and foreign key so it's easier to join sep31_historic with sep31

File: platform/src/main/resources/db/migration/V6__sep24_field_updates.sql [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+INSERT INTO sep31_historic (id, refunds)
+SELECT id, refunds
+FROM sep31_transaction;
+
+ALTER TABLE sep31_historic ADD CONSTRAINT FK_SEP31_REFUND FOREIGN KEY (id) REFERENCES sep31_transaction (id);
+
+
+ALTER TABLE sep31_transaction DROP COLUMN refunds;
+
+ALTER TABLE sep31_transaction ADD refunds JSON;
```
Comment (Ifropc) [inline platform/src/main/resources/db/migration/V6__sep24_field_updates.sql:70]:
```suggestion
ALTER TABLE sep31_transaction ALTER COLUMN refunds type JSONB using refunds::jsonb;
```

**Review by Ifropc — APPROVED** [review]

File: platform/src/main/resources/db/migration/V6__sep24_field_updates.sql [hunk 6]
```diff
@@ -0,0 +1,59 @@
+ALTER TABLE sep24_transaction ADD sep10_account_memo VARCHAR(255);
```
Comment (Ifropc) [inline platform/src/main/resources/db/migration/V6__sep24_field_updates.sql:1]: Should be renamed to v7?

**Review by lijamie98 — APPROVED** [review]

**Pull request merged**